util/start_test

```diff
@@ -582,17 +584,24 @@ def check_environment():
             print("Error: Cannot find {0}.".format(util_dir))
             sys.exit(1)
 
+def check_environment_with_args():
     # find test dir and check for access
     global test_dir, auto_gen_spec_tests
-    test_dir = os.path.join(home, "test")
+
     auto_gen_spec_tests = os.access(os.path.join(home, "spec"), os.F_OK)
-    if (not os.access(test_dir, os.F_OK) or not os.access(test_dir, os.W_OK)
-        or not os.access(test_dir, os.X_OK)):
-        test_dir = os.path.join(home, "examples")
-        auto_gen_spec_tests = False
+
+    if args.test_root_dir and re.match(r'/+[^/]', str(args.test_root_dir)):
+        test_dir = str(args.test_root_dir)
```

Comment:
If I'm reading this right, it will only match a value of 'test_root_dir' with an absolute path. Could we add a warning or error if args.test_root_dir is set but is not an absolute path? Otherwise it seems like things might just silently not do what the user expects.